File: 238-product-of-array-except-self/238-product-of-array-except-self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        p = 1
        n = len(nums)
        output = []
        for i in range(0,n):
            output.append(p)
            p *= nums[i]
        p = 1
        for i in range(n-1,-1,-1):
            output[i] *= p
            p *= nums[i]
        return output

File: 238-product-of-array-except-self/238-product-of-array-except-self.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
const productExceptSelf = (nums) => {
    const output = [];
    let left = 1, right = 1;
    for (let i = nums.length - 1; i >= 0; i--) {
        output[i] = right;
        right *= nums[i];
    }
    for (let j = 0; j < nums.length; j++) {
        output[j] *= left;
        left *= nums[j];
    }
    return output;
};